components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java

```diff
@@ -703,8 +712,15 @@ public boolean doPostUpdateCredentialByAdmin(String userName, Object credential,
             return true;
         }
         if (!userStoreManager.isReadOnly()) {
-            userStoreManager.setUserClaimValue(userName, IdentityMgtConstants.LAST_PASSWORD_UPDATE_TIME, Long
-                    .toString(System.currentTimeMillis()), null);
+            String domainName = userStoreManager.getRealmConfiguration().getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME);
+            String usernameWithDomain = UserCoreUtil.addDomainToName(userName, domainName);
+            boolean isUserExistInCurrentDomain = userStoreManager.isExistingUser(usernameWithDomain);
+            if (isUserExistInCurrentDomain) {
+                Map<String, String> userClaims = new HashMap<>();
+                userClaims.put(IdentityMgtConstants.LAST_PASSWORD_UPDATE_TIME, Long
+                        .toString(System.currentTimeMillis()));
+                userStoreManager.setUserClaimValues(userName, userClaims, null);
+            }
```

Comment:
Is the isUserExisting check needed for this post listener method ? This method will ideally be called only on the relevant UserStoreManager unlike authenticate method which loops through all the UserStoreManagers. So we don't need to check for user existence right ? It can be a very costly operation.